ouroboros-consensus/src/Ouroboros/Storage/FS/API/Types.hs

```diff
@@ -75,7 +82,38 @@ allowExisting openMode = case openMode of
   Paths
 -------------------------------------------------------------------------------}
 
-type FsPath = [String]
+newtype FsPath = UnsafeFsPath { fsPathToList :: [Strict.Text] }
+  deriving (Eq, Ord, Generic)
+
+fsPathFromList :: [Strict.Text] -> FsPath
+fsPathFromList = UnsafeFsPath . force
+
+instance Show FsPath where
+  show = intercalate "/" . map Strict.unpack . fsPathToList
+
+instance Condense FsPath where
+  condense = show
+
+-- | Constructor for 'FsPath' ensures path is in normal form
+mkFsPath :: [String] -> FsPath
+mkFsPath = fsPathFromList . map Strict.pack
```

Comment:
What about `OverloadedLists` to avoid changing all occurrences of `[..]` to `mkFsPath [..]`? 😁